tests/system/small/test_null_index.py

```diff
@@ -268,6 +268,25 @@ def test_null_index_series_self_join(
     )
 
 
+def test_null_index_series_self_join_on(
```

Comment:
were we able to write a sample code that reproduces the issue in b/359602477?  It would be great to add that as a test case `.map()`

Reply:
added test in new revision

Reply:
Thank you!